helix-term/src/main.rs

```diff
@@ -115,25 +115,69 @@ FLAGS:
 
     setup_logging(logpath, args.verbosity).context("failed to initialize logging")?;
 
-    let config_dir = helix_loader::config_dir();
-    if !config_dir.exists() {
-        std::fs::create_dir_all(&config_dir).ok();
-    }
-
     helix_loader::initialize_config_file(args.config_file.clone());
 
-    let config = match std::fs::read_to_string(helix_loader::config_file()) {
-        Ok(config) => toml::from_str(&config)
-            .map(helix_term::keymap::merge_keys)
-            .unwrap_or_else(|err| {
-                eprintln!("Bad config: {}", err);
-                eprintln!("Press <ENTER> to continue with default config");
-                use std::io::Read;
-                let _ = std::io::stdin().read(&mut []);
-                Config::default()
-            }),
-        Err(err) if err.kind() == std::io::ErrorKind::NotFound => Config::default(),
-        Err(err) => return Err(Error::new(err)),
+    // We need the config, to get the editor.security.[load|confirm]_local_config flags
+    let root_config: Config = std::fs::read_to_string(helix_loader::config_file())
+        .map(|config| toml::from_str(&config))?
+        .unwrap_or_default();
+
+    // Now, let's load config based on a PathBuf
+    // We will keep it toml::Value so we can use helix_loader::merge_toml_values
+    // This will respect editor.security settings
+    let load_config = |config_path: std::path::PathBuf| {
+        if std::path::Path::new(&config_path).exists() {
+            let mut confirmed = true;
+            if config_path != helix_loader::config_file() {
+                if root_config.editor.security.load_local_config {
+                    if root_config.editor.security.confirm_local_config {
+                        eprintln!(
+                            "Type yes to continue with loading config: {:#?}",
+                            config_path
+                        );
+                        use std::io::Read;
+                        confirmed = match std::io::stdin().read(&mut [1, 2, 3, 4]) {
+                            Ok(n) => n == "yesE".len(),
+                            Err(_) => false,
+                        }
```

Comment:
What is this doing?

Reply:
That whole thing is embarrasing. It should rather match on the actual input of `yes`. I thought I had fixed that to be honest. I'll make a pass on it now and refactor out that function you requested.